packages/flutter_tools/lib/src/device.dart

```diff
@@ -915,6 +915,42 @@ class DebuggingOptions {
 
   bool get hasObservatoryPort => hostVmServicePort != null;
 
+  List<String> getIOSLaunchArguments(EnvironmentType environmentType, String? route,  Map<String, Object?> platformArgs) {
+    final String dartVmFlags = computeDartVmFlags(this);
+    return <String>[
+      '--enable-dart-profiling',
+      if (environmentType == EnvironmentType.physical) '--disable-service-auth-codes',
+      if (environmentType == EnvironmentType.simulator && disableServiceAuthCodes) '--disable-service-auth-codes',
```

Comment:
Talked to @jonahwilliams, I'm pretty sure this logic is out of date.  For physical devices this was added in https://github.com/flutter/flutter/pull/49735/files#diff-5a3d215e39f292ac706d495f042ef45fab9f13962022f4e8e4580254e8557d52R322 to always be passed in, because there used to be a fallback where we suggested a port to use to the device, and the tool tried that port first, but didn't have the auth code.  However, that fallback was removed in https://github.com/flutter/flutter/pull/68756/files#diff-5a3d215e39f292ac706d495f042ef45fab9f13962022f4e8e4580254e8557d52L368

That means this can be:
```suggestion
      if (disableServiceAuthCodes) '--disable-service-auth-codes',
```
and if something unexpected breaks, we can revert to the old behavior.

Reply:
Changed in Changed in https://github.com/flutter/flutter/pull/114628/commits/c6b500e5dc835ccba5041ca51119424cba078c0c..